clients/src/main/java/org/apache/kafka/common/security/JaasConfig.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.security;
+
+import java.io.IOException;
+import java.io.StreamTokenizer;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import javax.security.auth.login.AppConfigurationEntry.LoginModuleControlFlag;
+
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.network.LoginType;
+
+class JaasConfig extends Configuration {
+
+    private final String loginContextName;
+    private final AppConfigurationEntry[] appConfig;
+
+    public JaasConfig(LoginType loginType, String jaasConfigParams) {
+        StreamTokenizer tokenizer = new StreamTokenizer(new StringReader(jaasConfigParams));
+        tokenizer.slashSlashComments(true);
+        tokenizer.slashStarComments(true);
+        tokenizer.wordChars('-', '-');
+        tokenizer.wordChars('_', '_');
+        tokenizer.wordChars('$', '$');
+
+        try {
+            List<AppConfigurationEntry> entries = new ArrayList<>();
+            while (tokenizer.nextToken() != StreamTokenizer.TT_EOF) {
+                entries.add(parseAppConfigurationEntry(tokenizer));
+            }
+            if (entries.isEmpty())
+                throw new IllegalArgumentException("Login module not specified in jaas config");
+
+            this.loginContextName = loginType.contextName();
+            this.appConfig = entries.toArray(new AppConfigurationEntry[entries.size()]);
+
+        } catch (IOException e) {
+            throw new KafkaException("Unexpected exception while parsing Jaas configuration");
```

Comment:
Even though the config is invalid, the passwords may be valid, so it seems safer not to include them. It would be nice if the sensitive entries in the JAAS config would be communicated in some way to improve debuggability. Btw, a nit: we seem to use inconsistent capitalisation of the word JAAS in our messages. It would be nice to make that consistent.